src/commands/UserManagement/onJoin.ts

```diff
@@ -0,0 +1,171 @@
+import type { GuildMemberRoleManager, Role, StringSelectMenuInteraction } from "discord.js";
+import { userMention } from "discord.js";
+import type { ArgsOf } from "discordx";
+import { Discord, On, SelectMenuComponent } from "discordx";
+import mustache from "mustache";
+
+import { JOIN_FORM_VALUES } from "../../data/constants";
+import { onJoinForm } from "../../data/forms";
+import MultiForm from "../../lib/discord/Prompt/MultiForm";
+import type { Properties } from "../../types/Utils";
+import deleteDiscordMessage from "../../utils/deleteDiscordMessage";
+
+@Discord()
+export class UserManagement {
```

Comment:
We should refactor the property _interaction and how the getters and setters for this property work so that we don't have to make so much type casting 

Ex:
`
 get memberRoleManager(): GuildMemberRoleManager {
    return this.interaction?.member?.roles as GuildMemberRoleManager;
}
`